File: codewars/cpp/perimeter_of_squares_in_a_rectangle.cpp
// https://www.codewars.com/kata/559a28007caad2ac4e000083/train/cpp


#include <iostream>

typedef unsigned long long ull;

class SumFct {
	public:
		static ull perimeter(int n) {
			ull f_term = 1;
			ull s_term = 1;
			ull n_term = 0;
			ull sum = f_term + s_term;

			for(int i = 2; i < n + 1; ++i) {
				n_term = f_term + s_term;
				sum += n_term;
				s_term = f_term;
				f_term = n_term;
			}

			return sum * 4;
		}
};

int main() {
	int n;

	std::cin >> n;

	SumFct out;

	std::cout << out.perimeter(n) << '\n';

	return 0;
}

File: codewars/perl/perimeter_of_squares_in_a_rectangle.pl
#!/usr/bin/perl -W

# https://www.codewars.com/kata/559a28007caad2ac4e000083/train/perl

package Solution;

use 5.30.0;
use strict;
use warnings;
use Exporter qw(import);
our @EXPORT_OK = qw(perimeter);

sub perimeter {
    my $n = $_[0];

    if ($n == 0) {
        return 4;
    }

    my $ft = 1;
    my $st = 1;
    my $nt = 0;
    my $s = $ft + $st;

    for my $i (2..$n) {
        $nt = $ft + $st;
        $st = $ft;
        $ft = $nt;
        $s = $s + $nt;
    }

   return $s * 4;
}

my $n = <>;

print(perimeter($n));
